### PR TITLE
Fix for hint count (dartbug.com/23395).

### PR DESCRIPTION
**Pull request by pq**
@bwilkerson 

Pardon the extra noise.  I did a little cleanup of the test grouping.

The salient bug is here: dartbug.com/23395.

### CODE REVIEW

**Comment by pq** [comment]
FWIW: @sethladd

File: test/mocks.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+library analyzer_cli.test.mocks;
+
+import 'package:analyzer/analyzer.dart';
+import 'package:analyzer/src/generated/engine.dart';
+import 'package:analyzer/src/generated/source.dart';
+import 'package:analyzer_cli/src/options.dart';
+import 'package:mockito/mockito.dart';
+
+
+class MockAnalysisError extends Mock implements AnalysisError {
```
Comment (bwilkerson) [inline test/mocks.dart:14]: I'm not sure how much sense it makes to create mocks for light-weight objects, like AnalysisError (and several others), but I don't have a strong objection to it either.

**Comment by bwilkerson** [comment]
LGTM

**Comment by sethladd** [comment]
Thanks!

**Pull request merged**